webview-ui/src/components/kilocode/common/DeviceAuthCard.tsx

```diff
@@ -0,0 +1,215 @@
+					{t("kilocode:deviceAuth.title")}
+				</h3>
+
+				{/* Step 1: URL Section */}
+				<div className="flex flex-col gap-3">
+					<p className="text-sm text-vscode-descriptionForeground text-center">
+						{t("kilocode:deviceAuth.step1")}
+					</p>
+
+					{/* URL Box with Copy and Open Browser */}
+					<div className="flex flex-col gap-2">
+						<div className="flex items-center gap-2 p-3 bg-vscode-input-background border border-vscode-input-border rounded">
+							<span className="flex-1 text-sm font-mono text-vscode-foreground truncate">
+								{verificationUrl}
+							</span>
+							<button
+								onClick={handleCopyUrl}
+								className="flex-shrink-0 p-1 hover:bg-vscode-toolbar-hoverBackground rounded"
+								title={t("kilocode:deviceAuth.copyUrl")}>
+								<span className="codicon codicon-copy text-vscode-foreground"></span>
+							</button>
+						</div>
+						<ButtonPrimary onClick={handleOpenBrowser}>
+							{t("kilocode:deviceAuth.openBrowser")}
+						</ButtonPrimary>
+					</div>
+
+					{/* QR Code Section */}
+					{qrCodeDataUrl && (
+						<div className="flex flex-col items-center gap-2 mt-2">
+							<p className="text-sm text-vscode-descriptionForeground">
+								{t("kilocode:deviceAuth.scanQr")}
+							</p>
+							<img
+								src={qrCodeDataUrl}
+								alt="QR Code"
+								className="w-40 h-40 border border-vscode-widget-border rounded"
+							/>
+						</div>
+					)}
+				</div>
+
+				{/* Step 2: Verification Section */}
+				<div className="flex flex-col gap-3">
+					<p className="text-sm text-vscode-descriptionForeground text-center">
+						{t("kilocode:deviceAuth.step2")}
+					</p>
+
+					{/* Verification Code */}
+					<div className="flex justify-center">
+						<div className="px-6 py-3 bg-vscode-input-background border-2 border-vscode-focusBorder rounded-lg">
+							<span className="text-2xl font-mono font-bold text-vscode-foreground tracking-wider">
+								{code}
+							</span>
+						</div>
+					</div>
+
+					{/* Time Remaining */}
+					<div className="flex items-center justify-center gap-2">
+						<span className="codicon codicon-clock text-vscode-descriptionForeground"></span>
+						<span className="text-sm text-vscode-descriptionForeground">
+							{t("kilocode:deviceAuth.timeRemaining", { time: formatTime(timeRemaining) })}
+						</span>
+					</div>
+
+					{/* Status */}
+					<div className="flex items-center justify-center gap-2">
+						<span className="codicon codicon-loading codicon-modifier-spin text-vscode-descriptionForeground"></span>
+						<span className="text-sm text-vscode-descriptionForeground">
+							{t("kilocode:deviceAuth.waiting")}
+						</span>
+					</div>
+				</div>
+
+				{/* Cancel Button */}
+				<div className="w-full flex flex-col">
+					<ButtonSecondary onClick={handleCancel}>{t("kilocode:deviceAuth.cancel")}</ButtonSecondary>
+				</div>
+			</div>
+		)
+	}
```

Comment:
I would make this a module scoped inner component in the same file, just to reduce the complexity of this outer component. not a blocker.